persist/fs/files.go

```diff
@@ -338,22 +358,54 @@ func commitlogFiles(commitLogsDir string, pattern string) ([]string, error) {
 }
 
 func filesBefore(files []string, t time.Time) ([]string, error) {
+	return filterFilesByTime(files, func(ft time.Time) (bool, bool) {
+		if ft.Before(t) {
+			// Include and continue scanning
+			return true, true
+		}
+
+		// Don't include and don't continue scanning because files
+		// are sorted by their timestamps in ascending order so we
+		// know we're done.
```

Comment:
Hm how are we sure that the files are guaranteed to be ordered by time?  It seems they’re just passed in here as an array.  Perhaps instead of passing an array of string we can pass an array of some struct that includes the time and then we can sort it explicitly to be safe/resilient to contracts between functions changing, etc.  Sort on presorted data will be v fast.